### PR TITLE
Limit the length of ES error messages

### PR DESCRIPTION
**Pull request by ndokos**
This is just to prevent a large error message from bloating the log
file. It does not solve the underlying problem which is that we sent
a payload that was bigger than the max that ES was willing to accept.

### CODE REVIEW

**Comment by ndokos** [comment]
This needs a unit test before it can be merged.

File: server/pbench/lib/pbench/__init__.py [hunk 1]
```diff
@@ -227,8 +228,9 @@ def actions_tracking_closure(cl_actions):
             status = resp['status']
         except KeyError as e:
             assert not ok
-            # resp is not of expected form
-            print(resp)
+            # resp is not of expected form.
+            # Limit the error message to 16K
+            print(resp[:_MAX_ERRMSG_LENGTH])
```
Comment (portante) [inline server/pbench/lib/pbench/__init__.py]: Do we have to first convert `resp` into a string to do this?  Perhaps we should use `json.dumps` instead?

And not sure we should put the size we are limiting the error message to in the comments.

Comment (ndokos) [reply server/pbench/lib/pbench/__init__.py]: Done in latest.

File: server/pbench/lib/pbench/__init__.py [hunk 2]
```diff
@@ -227,8 +228,10 @@ def actions_tracking_closure(cl_actions):
             status = resp['status']
         except KeyError as e:
             assert not ok
-            # resp is not of expected form
-            print(resp)
+            # resp is not of expected form.
+            # Limit the length of the error message.
+            msg = json.dumps(resp, indent=4)
+            print(msg[:_MAX_ERRMSG_LENGTH])
```
[the remark this answers is not in the document]
Comment (ndokos) [reply server/pbench/lib/pbench/__init__.py]: I'll do that, but just for my benefit: are you worried about a full string copy? If so, I'm pretty sure that that does not happen. It is just a ref.

File: server/pbench/lib/pbench/__init__.py [hunk 3]
```diff
@@ -248,8 +251,10 @@ def actions_tracking_closure(cl_actions):
                 errorsfp.flush()
                 failures += 1
             else:
-                # Retry all other errors
-                print(resp)
+                # Retry all other errors.
+                # Limit the length of the error message.
+                msg = json.dumps(resp, indent=4)
+                print(msg[:_MAX_ERRMSG_LENGTH])
```
[the remark this answers is not in the document]
Comment (ndokos) [reply server/pbench/lib/pbench/__init__.py]: Sure,

**Comment by ndokos** [comment]
@portante: did the changes and squashed, I think it's ready.

File: server/pbench/lib/pbench/__init__.py [hunk 4]
```diff
@@ -227,8 +228,9 @@ def actions_tracking_closure(cl_actions):
             status = resp['status']
         except KeyError as e:
             assert not ok
-            # resp is not of expected form
-            print(resp)
+            # resp is not of expected form.
+            # Limit the length of the error message.
+            print("es_index: WARNING - retrying error %r\n%s" % (e, json.dumps(resp, indent=4)[:_MAX_ERRMSG_LENGTH]), file=sys.stderr)
```
Comment (portante) [inline server/pbench/lib/pbench/__init__.py]: Do we retry these errors?

**Pull request merged**